app/view_models/base_with_adjustments.rb

```diff
@@ -1,19 +1,15 @@
 class BaseWithAdjustments < BaseViewModel
-  attribute :adjustments, default: []
-
-  def previous_explanation
-    field = adjustments.last
-    return unless field
-
-    field.details['comment']
+  attribute :adjustment_comment
+
+  def self.adjustable_attribute(attribute_name, type, options = {})
+    attribute attribute_name, type, **options
+    attribute :"#{attribute_name}_original", type, **options
+    define_method :"original_#{attribute_name}" do
+      attributes["#{attribute_name}_original"] || attributes[attribute_name.to_s]
```

Comment:
I don't think I like this... but that is because I think the naming will become confusing..

don't have any better suggestions - but will think on this. :)